dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/Map4.java

```diff
@@ -68,11 +69,11 @@ public V getValue( R key1, S key2, T key3, U key4 )
     }
 
     @SafeVarargs
-    public static <R, S, T, U, V> Map4<R, S, T, U, V> asMap4( final SimpleEntry<R, MapMapMap<S, T, U, V>>... entries )
+    public static <R, S, T, U, V> Map4<R, S, T, U, V> ofEntries( Map.Entry<R, MapMapMap<S, T, U, V>>... entries )
```

Comment:
This is getting a bit hard to wrap your head around :-) Could it be made into a model somehow?

Reply:
Actually the final ExpressionParserService code ended up not using Map4 at all, but I had made this refactor when I thought that it would. It aligns the code with the Java 9 Map.ofEntries syntax and datatypes, so it will be more familiar to future readers who know Java 9. Even though the new parser code doesn't use Map4, I left the edit in because I thought it improved the code.

But the point is well taken; this code is hard to comprehend. I'm not sure what you mean by a model; any suggestions? I will keep it in mind as well.